xwiki-platform-core/xwiki-platform-realtime/xwiki-platform-realtime-default/src/main/java/org/xwiki/realtime/DefaultRealtimeEditorManager.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.realtime;
+
+import java.util.List;
+
+import javax.annotation.Priority;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.container.Container;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.netflux.EntityChannel;
+import org.xwiki.netflux.EntityChannelStore;
+import org.xwiki.script.ScriptContextManager;
+import org.xwiki.script.service.ScriptService;
+import org.xwiki.sheet.SheetManager;
+import org.xwiki.wysiwyg.script.WysiwygEditorScriptService;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.doc.XWikiDocument;
+
+/**
+ * @version $Id$
+ */
+@Component
+@Singleton
+@Priority(1000)
+public class DefaultRealtimeEditorManager implements RealtimeEditorManager
+{
+    private static String editorKey = "editor";
+
+    private static String wysiwyg = "wysiwyg";
+
+    @Inject
+    private Provider<XWikiContext> xwikiContextProvider;
+
+    @Inject
+    private EntityChannelStore entityChannelStore;
+
+    @Inject
+    private SheetManager sheetManager;
+
+    @Inject
+    private ScriptContextManager scriptContextManager;
+
+    @Inject
+    @Named("wysiwyg")
+    private ScriptService wysiwygEditorScriptService;
+
+    @Inject
+    private Container container;
+
+    /**
+     * {@inheritDoc}
+     * 
+     * @see org.xwiki.realtime.RealtimeEditorManager#getSelectedEditor(org.xwiki.bridge.DocumentModelBridge)
+     */
+    public String getSelectedEditor()
+    {
+        // When there is an editor query parameter, it is the selected editor.
+        String requestEditor = "";
+        Object requestEditorObj = this.container.getRequest().getProperty(editorKey);
```

Comment:
Sounds like you could simplify with `String.valueOf(this.container.getRequest().getProperty(editorKey))`

Reply:
I think that `this.container.getRequest().getProperty(editorKey)` can return `null`, which would make `String.valueOf` raise a `NullPointerException`. I went with `StringUtils.isEmpty` instead :)